sqlparser/sqlparser.go

```diff
@@ -197,30 +202,186 @@ func (p *Parser) replaceInsertValueFromValArg(query *InsertQuery, colIndex int,
+		} else {
+			query.ColumnValues[colIndex] = createSQLIntTypeVal(*arg)
+		}
+	case *int32:
+		if colName == p.shardKeyColumnName(query.TableName) {
+			if arg == nil {
+				return errors.WithStack(ErrShardingKeyNotAllowNil)
+			}
+			query.ShardKeyID = Identifier(*arg)
+		}
+		if arg == nil {
+			query.ColumnValues[colIndex] = createSQLNilTypeVal()
+		} else {
+			query.ColumnValues[colIndex] = createSQLIntTypeVal(*arg)
+		}
+	case *int64:
+		if colName == p.shardKeyColumnName(query.TableName) {
+			if arg == nil {
+				return errors.WithStack(ErrShardingKeyNotAllowNil)
 			}
+			query.ShardKeyID = Identifier(*arg)
+		}
+		if arg == nil {
+			query.ColumnValues[colIndex] = createSQLNilTypeVal()
+		} else {
+			query.ColumnValues[colIndex] = createSQLIntTypeVal(*arg)
 		}
 	case uint, uint8, uint16, uint32, uint64:
 		if colName == p.shardKeyColumnName(query.TableName) {
 			query.ShardKeyID = Identifier(int64(arg.(uint64)))
 		}
+		query.ColumnValues[colIndex] = createSQLIntTypeVal(arg)
+	case *uint:
+		if colName == p.shardKeyColumnName(query.TableName) {
+			if arg == nil {
+				return errors.WithStack(ErrShardingKeyNotAllowNil)
+			}
+			query.ShardKeyID = Identifier(*arg)
+		}
+		if arg == nil {
+			query.ColumnValues[colIndex] = createSQLNilTypeVal()
+		} else {
+			query.ColumnValues[colIndex] = createSQLIntTypeVal(*arg)
+		}
+	case *uint8:
+		if colName == p.shardKeyColumnName(query.TableName) {
+			if arg == nil {
+				return errors.WithStack(ErrShardingKeyNotAllowNil)
+			}
+			query.ShardKeyID = Identifier(*arg)
+		}
+		if arg == nil {
+			query.ColumnValues[colIndex] = createSQLNilTypeVal()
+		} else {
+			query.ColumnValues[colIndex] = createSQLIntTypeVal(*arg)
+		}
+	case *uint16:
+		if colName == p.shardKeyColumnName(query.TableName) {
+			if arg == nil {
+				return errors.WithStack(ErrShardingKeyNotAllowNil)
+			}
+			query.ShardKeyID = Identifier(*arg)
+		}
+		if arg == nil {
+			query.ColumnValues[colIndex] = createSQLNilTypeVal()
+		} else {
+			query.ColumnValues[colIndex] = createSQLIntTypeVal(*arg)
+		}
+	case *uint32:
+		if colName == p.shardKeyColumnName(query.TableName) {
+			if arg == nil {
+				return errors.WithStack(ErrShardingKeyNotAllowNil)
+			}
+			query.ShardKeyID = Identifier(*arg)
+		}
+		if arg == nil {
+			query.ColumnValues[colIndex] = createSQLNilTypeVal()
+		} else {
+			query.ColumnValues[colIndex] = createSQLIntTypeVal(*arg)
+		}
+	case *uint64:
+		if colName == p.shardKeyColumnName(query.TableName) {
+			if arg == nil {
+				return errors.WithStack(ErrShardingKeyNotAllowNil)
+			}
+			query.ShardKeyID = Identifier(*arg)
+		}
+		if arg == nil {
+			query.ColumnValues[colIndex] = createSQLNilTypeVal()
+		} else {
+			query.ColumnValues[colIndex] = createSQLIntTypeVal(*arg)
+		}
+	case bool:
+		val := convertBoolToInt8(arg)
+		query.ColumnValues[colIndex] = createSQLIntTypeVal(val)
+	case *bool:
+		if arg == nil {
+			query.ColumnValues[colIndex] = func() *vtparser.SQLVal {
```

Comment:
This section can refactor by `createSQLNilTypeVal`

Reply:
https://github.com/knocknote/octillery/pull/17/commits/cf85cdc8ca29bed49254907192b1624246b5c5fd